conduit/pipeline/pipeline.go

```diff
@@ -64,6 +63,9 @@ type pipelineImpl struct {
 	completeCallback []conduit.OnCompleteFunc
 
 	pipelineMetadata state
+
+	statusMu sync.Mutex
```

Comment:
With this change, can we wait until after #128 goes in before merging this PR? It may require an additional EC2 test.

Reply:
Sure. Adding a mutex was much less disruptive than switching to `atomic.Uint64`.